llama.cpp

```diff
@@ -14648,6 +14818,30 @@ LLAMA_API int32_t llama_chat_apply_template(
     return res;
 }
 
+LLAMA_API int llama_split_path(char * split_path, size_t maxlen, const char * path_prefix, int split_no, int split_count) {
+    static const char * const SPLIT_PATH_FORMAT = "%s-%05d-of-%05d.gguf";
+    if (snprintf(split_path, maxlen, SPLIT_PATH_FORMAT, path_prefix, split_no + 1, split_count)) {
+        return strlen(split_path);
+    }
+    return 0;
+}
+
+int llama_split_prefix(char * dest, size_t maxlen, const char * split_path, int split_no, int split_count) {
+    std::string str_split_path(split_path);
+    char postfix[32];
+    sprintf(postfix, "-%05d-of-%05d.gguf", split_no + 1, split_count);
+    std::string str_postfix(postfix);
+
+    // check if dest ends with postfix
+    auto size_prefix = str_split_path.size() - str_postfix.size();
+    if (size_prefix > 0 && str_split_path.find(str_postfix, size_prefix) != std::string::npos) {
+        strncpy(dest, split_path, std::min(size_prefix, maxlen));
```

Comment:
`strncpy` is generally not safe to use for this purpose because it does not ensure that the string is NUL-terminated. A good alternative is `snprintf` with `%s` format.

Reply:
I pushed the fix on https://github.com/ggerganov/llama.cpp/pull/6187/commits/764c7afee781c2d2d1966df6214073d467446767 . Thanks for noticing, I was making this function at late night so I didn't really think much about it.